resource/js/hierarchy.js

```diff
@@ -456,8 +553,35 @@ function getTreeConfiguration() {
                 else return -1;
             }
             else if (bNotation) return 1;
-        }
-        return naturalCompare(aNode.text.toLowerCase(), bNode.text.toLowerCase());
+        } else {
```

Comment:
This seems a bit overly complex. I tested replacing the whole block with this:

```js
        } else {
          // no sorting on notation requested
          // make sure the tree nodes with class 'domain' are sorted before the others
          // aDomain/bDomain will be "0" if a/b has a domain class, else 1
          var aDomain = (int) (aNode.original.a_attr['class'] != 'domain');
          var bDomain = (int) (bNode.original.a_attr['class'] != 'domain');
          return naturalCompare(aDomain + " " + aNode.text.toLowerCase(),
                                bDomain + " " + bNode.text.toLowerCase());
        }
```

It gave me the same result.

Reply:
Didn't you get any error due to the `(int)` casting? I normally use `parseInt()` or `Number` in JS/TS, and **think** that's still not available in ES6. But tested in a codepen [pen](https://codepen.io/kinow/pen/wvoRwym?editors=1111) just to make sure.

![image](https://user-images.githubusercontent.com/304786/110675498-d3040400-8237-11eb-9e73-19a8b4dcbb92.png)

Reply:
Ah sorry, my bad! You are of course right @kinow, I was mixing up Java syntax with JS. Just shows how little I do JS coding :) And indeed, `showNotation` was enabled because that is the default setting for vocabularies, so execution never went to that `else` block.

Here is a better fix that I also tested, double-checking that it does get executed (I also checked that if I invert the logic here, the domains will be sorted last):

```js
        } else {
          // no sorting on notation requested
          // make sure the tree nodes with class 'domain' are sorted before the others
          // aDomain/bDomain will be "0" if a/b has a domain class, else "1"
          var aDomain = (aNode.original.a_attr['class'] == 'domain') ? "0" : "1";
          var bDomain = (bNode.original.a_attr['class'] == 'domain') ? "0" : "1";
          return naturalCompare(aDomain + " " + aNode.text.toLowerCase(),
                                bDomain + " " + bNode.text.toLowerCase());
        }
```